agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java

```diff
@@ -78,6 +79,7 @@ private TelemetryItem createMessageTelemetryItem(LogData log) {
     setTime(telemetryBuilder, log.getEpochNanos());
     setSampleRate(telemetryBuilder, log);
     setExtraAttributes(telemetryBuilder, attributes);
+    ResourceParser.updateRoleNameAndInstance(telemetryBuilder, log.getResource());
```

Comment:
for the Javaagent I think this is always going to be `unknown_service:java` and overwrite the configured role name?

Reply:
did you mean to resolve this comment? I'm guessing this may be why some smoke tests are failing

Reply:
osvaldo said don't send ai.role.name when service.name is not set, i.e. don't send `unknown_service`. `:java` is the `process.execution.name`. you're correct, it's not a bug on opentelemetry-java sdk.

Reply:
tests are failing because test data has null resource.

Reply:
https://github.com/microsoft/Telemetry-Collection-Spec/pull/78#discussion_r898282377

Reply:
got it.. populateDefaults is called before update.. this line will override the customer's configuration.